opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/RelevanceQuery.java

```diff
@@ -31,26 +27,52 @@
 
   @Override
   public QueryBuilder build(FunctionExpression func) {
-    List<Expression> arguments = func.getArguments();
+    var arguments = func.getArguments().stream()
+        .map(a -> (NamedArgumentExpression)a).collect(Collectors.toList());
     if (arguments.size() < 2) {
       throw new SyntaxCheckException(
           String.format("%s requires at least two parameters", getQueryName()));
     }
-    NamedArgumentExpression field = (NamedArgumentExpression) arguments.get(0);
-    NamedArgumentExpression query = (NamedArgumentExpression) arguments.get(1);
-    T queryBuilder = createQueryBuilder(field, query);
 
-    Iterator<Expression> iterator = arguments.listIterator(2);
-    Set<String> visitedParms = new HashSet();
+    // Aggregate parameters by name, so getting a Map<Name:String, List>
+    arguments.stream().collect(Collectors.groupingBy(a -> a.getArgName().toLowerCase()))
+        .forEach((k, v) -> {
+          if (v.size() > 1) {
+            throw new SemanticCheckException(
+                String.format("Parameter '%s' can only be specified once.", k));
+          }
+        });
+
+    // Extract 'field'/'fields' and 'query'
+    var field = arguments.stream()
+        .filter(a -> a.getArgName().equalsIgnoreCase("field"))
+        .findFirst()
+        .orElse(null);
+    var fields = arguments.stream()
+        .filter(a -> a.getArgName().equalsIgnoreCase("fields"))
+        .findFirst()
+        .orElse(null);
+    if (fields == null && field == null) {
+      throw new SemanticCheckException("Both 'field' and 'fields' parameters are missing.");
+    }
+    if (fields != null && field != null) {
+      throw new SemanticCheckException("Both 'field' and 'fields' parameters are given.");
+    }
+
+    var query = arguments.stream()
+        .filter(a -> a.getArgName().equalsIgnoreCase("query"))
+        .findFirst()
+        .orElseThrow(() -> new SemanticCheckException("'query' parameter is missing"));
+    T queryBuilder = createQueryBuilder((field != null ? field : fields), query);
```

Comment:
Code specific to single field functions belongs in `SingleFieldQuery` class, multi-field functions -- `MultiFieldQuery`.

Reply:
Thank you, fixed in 2f1e3022.
Notes:
1. we can't check whether `field` and `fields` present both
2. `query` check could be moved to a separate method to the superclass, but the benefit is rather low